src/pages/home/Home.jsx

```diff
@@ -0,0 +1,127 @@
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchMovieData();
+  }, []);
+
+  const handleSearch = (event) => {
+    setSearchQuery(event.target.value);
+  };
+
+  const searchMovie = async () => {
+    try {
+      const response = await fetch(
+        `https://api.themoviedb.org/3/search/movie?query=${searchQuery}&api_key=${apiKey}&include_adult=false&language=en-US&page=1`
+      );
+      if (response.ok) {
+        const data = await response.json();
+        if (data.results.length > 0) {
+          const movieId = data.results[0].id;
+          return `/movie/${movieId}`;
+        } else {
+          setSearchError(true);
+          throw new Error("Failed at fetch the movie list");
+        }
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const handleSearchEnter = async (event) => {
+    if (event.key === "Enter") {
+      event.preventDefault();
+      const movieLink = await searchMovie();
+      console.log(movieLink);
+      if (movieLink) {
+        window.location.href = movieLink;
+      }
+    }
+  };
+
+  return (
+    <>
+      <TopMenu />
+      <section className="home-section">
+        <div className="search-movie-container">
+          <form className="search-bar">
+            <label htmlFor="movie-search" className="search-bar-label">
+              Which movie do you want to see?
+            </label>
+            <div className="input-container">
+              <span className="search-icon">
+                <img src="./search.svg" alt="search-icon" />
+              </span>
+              <input
+                className="search-bar-input"
+                type="text"
+                id="movie-search"
+                name="movie-search"
+                value={searchQuery}
+                onChange={handleSearch}
+                onKeyDown={handleSearchEnter}
+                placeholder="Search by name..."
+              />
+              <div className="search-error-message-container">
+                {searchError && (
+                  <p className="search-error-message">
+                    Can&#39;t find any result. Please check the name again...
+                  </p>
+                )}{" "}
+              </div>
+            </div>
+          </form>
+        </div>
+        {isLoading ? (
+          <p className="loading-text">Loading...</p>
+        ) : (
+          <Link to="/movie/now_playing">
+            <div className="scoller">
+              <ul className="home-carousel-list scoller-inner">
+                {movieHomeData.results.map((movie) => (
+                  <li key={movie.id}>
+                    <img
+                      src={`http://image.tmdb.org/t/p//w185${movie.poster_path}`}
+                      alt={movie.title}
+                      loading="lazy"
+                    />
+                  </li>
+                ))}
+              </ul>
+            </div>
+          </Link>

```

Comment:
Really cool carousel! Just remember to think about spelling, as it can cause bugs if we're not careful

Reply:
Thank you for the reminder!!